pagefetcher.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python2
+
+import urllib
+from os import path, makedirs
+
+"""
+Description:
+
+    This program is a module for wiphisher.py file which includes functionality
+    for downloading extra templates.
+
+Copyright:
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>
+"""
+__author__ = "blackHatMonkey"
+__license__ = "GPL"
+__version__ = "0.1"
+__status__ = "Development"
+__email__ = "dark.world@riseup.net"
+__maintainer__ = "blackHatMonkey"
+
+
+class UrlNotAvailable(Exception):
+    """ Exception class to raise in case of a bad URL. """
+
+    def __init__(self):
+        Exception.__init__(self, "The given URL is not available!")
+
+
+class TemplateNotAvailable(Exception):
+    """ Exception class to raise in case of non-existence template. """
+
+    def __init__(self):
+        Exception.__init__(self, "The given template is not available!")
+
+
+def fetch(template):
+    """
+    Fetch all the files for the given template.
+
+    Args:
+        template (str): The name of requested template.
+
+    Returns:
+        None
+
+    Raises:
+        TemplateNotAvailble: If the given template is not in the database.
+    """
+
+    # check if given directory exists and if not create it
+    if not path.isdir(template):
+        makedirs(template)
```

Comment:
If the directory exists, that means the user most likely has already downloaded the template. The execution should stop here.